src/primitives/uint.ts

```diff
@@ -12,10 +27,28 @@ export class UnsignedInteger {
     return false;
   }
 
+  static toUint32(num: number): Uint32 {
+    if (this.isUint32(num)) {
+      return num;
+    }
+    throw new TypeError();
+  }
+
   static toUint64(num: bigint): Uint64 {
     if (this.isUint64(num)) {
       return num;
     }
     throw new TypeError();
   }
+
+  static upCast(num: Uint32): Uint64 {
+    return BigInt(num) as Uint64;
+  }
+
+  static downCast(num: Uint64): Uint32 {
+    if (num > 0x100000000n) {
+      throw new TypeError();
+    }
+    return Number(num) as Uint32;
```

Comment:
```suggestion
    return num as Uint32;
```

I think should work

Reply:
This doesn't work because in the case `num` is a `bigint` so I need to use the Number constructor.

Reply:
Ah ok, then lgtm